### PR TITLE
347 Dark Mode initial implementation

### LINKED ISSUE #347: Add Dark Mode button and change the 'Me' page accordingly
Users frequently find themselves in situations where they are using the app in very bright sunlight, or at night. Please add a Dark Mode button, and change the 'Me' page accordingly. 

This is the screen when the user clicks their avatar on the home screen. 

The screen should look like this - https://mongodb.invisionapp.com/share/PCYZ6ZW7FR8#/screens/433591779_Me
Dark mode is off by default, and has the light background; when Dark mode is toggled on, the dark theme should be applied. 

Color key for light mode: https://mongodb.invisionapp.com/share/PCYZ6ZW7FR8#/screens/433591768_Color_Key_-_Light_Mode  
Color key for dark mode: https://mongodb.invisionapp.com/share/PCYZ6ZW7FR8#/screens/433591769_Color_Key_-_Dark_Mode   
Component Color specs for light mode: https://mongodb.invisionapp.com/share/PCYZ6ZW7FR8#/screens/433591770_Component_Colors_-_Light_Mode
Component Color specs for dark mode: https://mongodb.invisionapp.com/share/PCYZ6ZW7FR8#/screens/433591771_Component_Colors_-_Dark_Mode

Both dark and light modes should follow the color key and specs above (it is possible that the current light mode is not).

This profile setting should be stored on the device, not in the database. 

Dark mode should work either when the switch is toggled on, or when the switch is off but the device itself is in dark mode.

Please make sure to TEST your build on a device or emulator to verify everything has changed - you can get a login and realm-app-id with easy instructions at https://wildaid.github.io/contribute/sandbox.html

### PR DESCRIPTION
**Pull request by bladebunny**
## Related Issue

Fixes #347
Changing the mode on the profile page now changes the color scheme. I have not changed all colors. Some views will auto-change based on the system colors. But we will need to go through and ensure views are using a correct color set color that is based on the design instead if relying on defaults.

## Checklist:

- [x] I have read the [contributor's guide](https://wildaid.github.io/contribute/index.html).
- [x] I linked an issue in the previous section
- [x] I have commented on the linked issue
- [x] I was assigned the linked issue (not required)
- [x] I have tested the change to the best of my ability against the [sandbox](https://wildaid.github.io/contribute/sandbox.html) or a [local build](https://wildaid.github.io/build).

Optional items:

- [x] My change adds new text and requires a change to translations.
- [x] My change requires a change to the documentation.
- [ ] I have submitted a PR to the [documentation repo](https://github.com/WildAid/wildaid.github.io).
- [ ] I was not able to test... (explain below, e.g. you did not have permissions to test a specific feature)
- [ ] This change depends O-FISH Realm repository changes (explain below)
- [ ] This change depends O-FISH Web repository changes (explain below)

* **Optional: Add any explanations here** 
I've backed out my additional localization stuff and this now only uses pre-existing loc set-up.
This PR can be closed once it passes merge checks and review.

* **Optional: Add any relevant screenshots here**

### CODE REVIEW

**Comment by o-fish-wildaid** [comment]
[Review PRs](https://trello.com/c/hsyA27BP/423-review-prs)

File: o-fish-ios/ViewModel/User/UserSettings.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+//
+//  UserSettings.swift
+//  
+//  Created on 10/9/20.
+//  Copyright © 2020 WildAid. All rights reserved.
+//
+
+import Foundation
```
Comment (am-MongoDB) [inline o-fish-ios/ViewModel/User/UserSettings.swift:8]: Remove?

**Review by am-MongoDB — CHANGES_REQUESTED** [review]
Could you please merge with the latest `main` when you make the changes (as it contains a fix for a bug that's preventing me from testing this PR.

Thanks.

**Comment by bladebunny** [comment]
Feedback incorporated and merge conflict in project file fixed.

File: o-fish-ios/ViewModel/User/UserSettings.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import Combine
+
+private struct StringKeys {
+    static let showDarkMode = "showDarkMode"
```
[the remark this answers is not in the document]
Comment (bladebunny) [reply o-fish-ios/ViewModel/User/UserSettings.swift]: Done

File: o-fish-ios/Helpers/String+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//  
+//  Created on 10/9/20.
+//  Copyright © 2020 WildAid. All rights reserved.
+//
+
+import Foundation
+
+public func NSLocalizedString(_ key: String) -> String {
+    return NSLocalizedString(key, comment: key)
+}
```
[the remark this answers is not in the document]
Comment (bladebunny) [reply o-fish-ios/Helpers/String+Extensions.swift]: I was using this earlier - removed now.

File: o-fish-ios/Helpers/String+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+extension Optional where Wrapped == String {
+
+    public var isNilOrEmpty: Bool {
+        if let value = self {
+            return (value.count == 0)
+        }
+
+        return true
+    }
```
[the remark this answers is not in the document]
Comment (bladebunny) [reply o-fish-ios/Helpers/String+Extensions.swift]: Removed.  Makes nil/empty checking strings easier but removed.

File: o-fish-ios/ViewModel/User/UserSettings.swift (hunk 1 above)
Comment (bladebunny) [reply o-fish-ios/ViewModel/User/UserSettings.swift:8]: Need to keep this.

File: o-fish-ios/Helpers/String+Extensions.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (am-MongoDB) [reply o-fish-ios/Helpers/String+Extensions.swift]: Thanks – I didn't see it used anywhere

**Review by am-MongoDB — CHANGES_REQUESTED** [review]
Still can't test, as this has been rebased onto `main`, but then the effects of https://github.com/WildAid/o-fish-ios/pull/363 have been undone (e.g. `DoneTextField.swift` has been added back in)

**Comment by bladebunny** [comment]
Sorry. I'll take a look. When I pulled there was a merged conflict that looked like it was adding DoneTextField. Will see if I can sort it out.

**Review by am-MongoDB — CHANGES_REQUESTED** [review]
It's still undoing part of the fix that went into `main` - e.g., these changes to the README

**Comment by am-MongoDB** [comment]
> It's still undoing part of the fix that went into `main` - e.g., these changes to the README

@bladebunny - actually, looks like the README is the only thing that needs backing out, so I should be able to test this version. Thanks.

**Comment by am-MongoDB** [comment]
I'll merge as-is and then fix the README afterwards

**Pull request merged**